In [1]:
import math

In [2]:
from scipy import stats

In [3]:
# Chi-Square Test

In [4]:
observed = [
    [50, 70],   # Very Satisfied
    [80, 100],  # Satisfied
    [60, 90],   # Neutral
    [30, 50],   # Unsatisfied
    [20, 50]    # Very Unsatisfied
]

In [5]:
print("\n--- Chi-Square Test for Independence ---")


--- Chi-Square Test for Independence ---


In [6]:
# Step 1: Hypotheses
print("Step 1: Hypotheses")
print("  H0: Device type and customer satisfaction are independent.")
print("  H1: Device type and customer satisfaction are not independent.")

Step 1: Hypotheses
  H0: Device type and customer satisfaction are independent.
  H1: Device type and customer satisfaction are not independent.


In [7]:
# Step 2: Calculate expected values and chi-square
row_totals = [sum(row) for row in observed]
col_totals = [sum(col) for col in zip(*observed)]
grand_total = sum(row_totals)

In [8]:
# Calculate expected frequencies
expected = []
for i in range(len(observed)):
    row_expected = []
    for j in range(len(observed[0])):
        e = row_totals[i] * col_totals[j] / grand_total
        row_expected.append(e)
    expected.append(row_expected)

In [9]:
# Compute chi-square statistic
chi_square = 0
for i in range(len(observed)):
    for j in range(len(observed[0])):
        o = observed[i][j]
        e = expected[i][j]
        chi_square += (o - e) ** 2 / e

In [10]:
# Degrees of freedom
dof = (len(observed) - 1) * (len(observed[0]) - 1)

In [11]:
# Step 3: Critical value
alpha = 0.05
critical_value = stats.chi2.ppf(1 - alpha, dof)

In [12]:
# Step 4: Decision
print("\nStep 2: Chi-Square Statistic = {:.4f}".format(chi_square))
print("Step 3: Critical Value (α=0.05, df={}) = {:.4f}".format(dof, critical_value))
if chi_square > critical_value:
    print("Step 4: Reject H0 → There is a significant association.")
else:
    print("Step 4: Fail to Reject H0 → No significant association.")
print("Explanation: The chi-square value is smaller than the critical value, so satisfaction does not depend on device type.\n")


Step 2: Chi-Square Statistic = 5.6382
Step 3: Critical Value (α=0.05, df=4) = 9.4877
Step 4: Fail to Reject H0 → No significant association.
Explanation: The chi-square value is smaller than the critical value, so satisfaction does not depend on device type.



Chi-Square Test:
The chi-square value (≈ 5.64) is smaller than the critical value (≈ 9.49). This means we fail to reject H₀ → customer satisfaction does not depend on device type.

In [13]:
# Part 2: Hypothesis Test

In [14]:
print("--- Hypothesis Test for Weekly Costs ---")

--- Hypothesis Test for Weekly Costs ---


In [15]:
# Step 1: Hypotheses
print("Step 1: Hypotheses")
print("  H0: The weekly cost follows the model (μ = 4000).")
print("  H1: The weekly cost is higher than the model (μ > 4000).")


Step 1: Hypotheses
  H0: The weekly cost follows the model (μ = 4000).
  H1: The weekly cost is higher than the model (μ > 4000).


In [16]:
# Given values
sample_mean = 3050     # from data
n = 25
x_mean = 600           # units produced
sigma_x = 25

In [17]:
# Theoretical mean and standard deviation
theoretical_mean = 1000 + 5 * x_mean   # W = 1000 + 5X
sigma_w = 5 * sigma_x                  # standard deviation
se = sigma_w / math.sqrt(n)            # standard error

In [ ]:
# Theoretical mean and standard deviation
theoretical_mean = 1000 + 5 * x_mean   # W = 1000 + 5X
sigma_w = 5 * sigma_x                  # standard deviation
se = sigma_w / math.sqrt(n)            # standard error

# Step 2: Test statistic
z = (sample_mean - theoretical_mean) / se

# Step 3: Critical value (z table, one-tailed, α=0.05)
critical_z = stats.norm.ppf(1 - alpha)

# Step 4: Decision
print("\nStep 2: Z Statistic = {:.4f}".format(z))
print("Step 3: Critical Z (α=0.05, one-tailed) = {:.4f}".format(critical_z))

if z > critical_z:
    print("Step 4: Reject H0 → Costs are significantly higher.")
else:
    print("Step 4: Fail to Reject H0 → No evidence that costs are higher.")

print("Explanation: The sample mean is lower than the theoretical mean, so there’s no proof costs increased.\n")

Hypothesis Test:
The Z value (≈ -38.0) is far below the critical value (≈ 1.65). This means we fail to reject H₀ → no evidence that weekly costs are higher than the model predicts.